python_modules/dagster/dagster/core/storage/event_log/sqlite/sqlite_event_log.py

```diff
@@ -230,9 +230,11 @@ def store_event(self, event):
         if event.is_dagster_event and event.dagster_event.asset_key:
             check.invariant(
                 event.dagster_event_type == DagsterEventType.ASSET_MATERIALIZATION
-                or event.dagster_event_type == DagsterEventType.ASSET_OBSERVATION,
-                "Can only store asset materializations and observations in index database",
+                or event.dagster_event_type == DagsterEventType.ASSET_OBSERVATION
+                or event.dagster_event_type == DagsterEventType.ASSET_MATERIALIZATION_PLANNED,
+                "Can only store asset materializations, materialization_planned, and observations in index database",
             )
+
             # mirror the event in the cross-run index database
             with self.index_connection() as conn:
                 conn.execute(insert_event_statement)
```

Comment:
<!-- GRAPHITE_METADATA {"trueLocation":{"startLine":235,"endLine":239}} -->
<i class='graphite__hidden'>[Re: lines 235 to 239]<br/><br/></i>

why remove instead of update?

Reply:
No strong reason here--I removed it earlier because I didn't feel that the check was necessary. I've updated it now though to contain the new event type